doozerlib/distgit.py

```diff
@@ -331,146 +331,110 @@ def image_build_method(self):
         odcs:
             packages:
-                mode: auto | manual (default)
-                # auto - detect packages in Dockerfile and merge with list below
-                # manual - only use list below. Ignore Dockerfile
-                exclude:  # when using auto mode, exclude these
-                  - package1
-                  - package2
+                mode: auto (default) | manual
+                # auto - If container.yaml with packages is given from source, use them.
+                #        Otherwise all packages with from the Koji build tag will be included.
+                # manual - only use list below
                 list:
                   - package1
                   - package2
-            platforms:
-                mode: auto | manual
-            config: ... # verbatim container.yaml content (see https://mojo.redhat.com/docs/DOC-1159997)
+        arches: # Optional list of image specific arches. If given, it must be a subset of group arches.
+          - x86_64
+          - s390x
+        container_yaml: ... # verbatim container.yaml content (see https://mojo.redhat.com/docs/DOC-1159997)
         """
 
-        arches = self.metadata.runtime.arches
-        if 'arches' in self.metadata.config:
-            arches = []
-            # only include arches from metadata that are
-            # valid globally
-            for a in self.metadata.config.arches:
-                if a in self.metadata.runtime.arches:
-                    arches.append(a)
-
-        if not self.runtime.odcs_mode:
-            config = {
-                'platforms': {'only': arches}
-            }
-            return config
-
-        no_source = self.config.content.source.alias is Missing
-        CYAML = 'build_container.yaml' if no_source else 'container.yaml'
-
-        # always delete from distgit
-        if os.path.exists(CYAML):
-            os.remove(CYAML)
-
-        if no_source:
-            source_container_yaml = CYAML
-        else:
-            source_container_yaml = os.path.join(self.source_path(), CYAML)
-        if os.path.isfile(source_container_yaml):
-            with open(source_container_yaml, 'r') as scy:
-                source_container_yaml = yaml.full_load(scy)
+        # list of platform (architecture) names to build this image for
+        arches = []  # type: list
+        global_arches = set(self.metadata.runtime.arches)
+        image_specific_arches = set(self.metadata.config.arches)
+        if not image_specific_arches:
+            # assume all global arches if no image specific arches are defined
+            arches = list(global_arches)
         else:
-            source_container_yaml = {}
+            # only include arches from metadata that are valid globally
+            missing_arches = image_specific_arches - global_arches
+            if missing_arches:
+                raise ValueError("Image specific arches ({}) are not enabled in group.yml.".format(", ".join(missing_arches)))
```

Comment:
note this now raises an error if you put an arch in an image that isn't in the group, where it would have silently ignored it before. i think this is the correct approach (and ISTR there's something else catching that use case anyway), just want that to be visible to anyone looking back on this PR.

that said, it _would_ be useful in some cases to go back to silent ignorance. this comes up when we disable an arch in the group.yml and don't want to have to go around and remove it from the individual images as well (especially if we plan to re-enable it shortly).